We can't test directly MAX Serving examples in our CI environment
Because we already execute workflow steps inside a CI container
Running another container (MAX Serving) "inside" a container (CI) has some complications.
The following steps are needed to make it work properly:
1. Fist we get a bridge network that GitHub Actions uses to run the CI container. GH runs containers in a separate bridge network that provides isolation from containers that aren't connected to that bridge network.
2. We start our MAX Serving container in the GitHub bridge network so that the CI container later has access to MAX Serving server.
3. We expose 8000 port because we use the bridge network not the host one.
4. We mount the host directory (not a directory on CI container) to our MAX Serving container because CI container directories aren't available for MAX Serving container.
The HOST_MODEL_REPOSITORY env var is set in the workflow because it uses a GitHub Actions macro.
5. We get the container IP address once it started. We use it later while querying for readiness and in Triton inference client.

diff --git a/SDK/public/max-repo/examples/inference/bert-python-torchscript/deploy.sh b/SDK/public/max-repo/examples/inference/bert-python-torchscript/deploy.sh
index 6c02c174a5..6572f45049 100755
--- a/SDK/public/max-repo/examples/inference/bert-python-torchscript/deploy.sh
+++ b/SDK/public/max-repo/examples/inference/bert-python-torchscript/deploy.sh
@@ -40,24 +40,27 @@ cp -r ${MODEL_PATH} model-repository/${MODEL_DIR}/1/
 cp config.pbtxt model-repository/${MODEL_DIR}/

 echo "Starting container"
+CONTAINER_BRIDGE_NETWORK=$(docker network ls --format "{{.Name}}" | grep "github_network_")
 CONTAINER_ID=$(\
-  docker run --rm -d --net=host \
-    -v $(pwd)/model-repository:/model-repository \
+  docker run --rm -d \
+    --network ${CONTAINER_BRIDGE_NETWORK} \
+    -p 8000:8000/tcp \
+    -v ${HOST_MODEL_REPOSITORY}:/model-repository \
     public.ecr.aws/modular/max-serving \
     tritonserver --model-repository=/model-repository \
     --load-model=${MODEL_DIR} \
 )
+CONTAINER_IP=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $CONTAINER_ID)

 printf "Compiling ${MODEL_DIR} model"
-until curl --output /dev/null --silent --fail localhost:8000/v2/health/ready; do
+until curl --output /dev/null --silent --fail ${CONTAINER_IP}:8000/v2/health/ready; do
     printf '.'
     sleep 5
 done
 printf "\nMAX Serving container started\n\n"

-
 # Execute the model with example input
-python3 triton-inference.py --input "$INPUT_EXAMPLE"
+python3 triton-inference.py --input "$INPUT_EXAMPLE" --url ${CONTAINER_IP}:8000

 printf "\nStopping container..."
 docker stop $CONTAINER_ID > /dev/null
